h/auth/util.py

```diff
@@ -54,6 +55,29 @@ def group_principals(user):
     return ['group:{group.pubid}'.format(group=group) for group in user.groups]
 
 
+def has_permission(request, context, userid, permission):
+    """
+    Return True if userid has permission on context object.
+
+    Return True if the given userid has the given permission on the given
+    context object, False otherwise.
+
+    For example:
+
+        if has_permission(request,
+                          some_annotation,
+                          'acct:philip@hypothes.is',
+                          'read'):
+            print 'philip can read this annotation'
+        else:
+            print "philip isn't allowed to read this annotation"
+
+    """
+    policy = request.registry.queryUtility(authorization.IAuthorizationPolicy)
+    users_principals = effective_principals(userid, request)
+    return bool(policy.permits(context, users_principals, permission))
```

Comment:
I'll wait to see if this is what @nickstenning was asking for when he asked for this code to be simplified (full commit for this bit is <https://github.com/hypothesis/h/pull/3238/commits/b8c4d1061a1b6cacb8c3deee7c4cd47e2411a557>), if it's good then I'll add tests for it.

I think mocking the authorization policy might makes tests for this bit of code kind of pointless - "It retrieves the (mock) authorization policy; It calls the (mock) effective principals; It calls permits; It returns True or False if the mock permits() returns truthy or falsey" - that seems like just describing the code line by line not testing it. So maybe I'll test against Pyramid's real `ACLAuthorizationPolicy`.